python/test/gluon/test_core.py

```diff
@@ -124,7 +124,7 @@ def warpgroup_mma_kernel(a, b, out, M: ttgl.constexpr, N: ttgl.constexpr, K: ttg
     acc = hopper.warpgroup_mma(a_shmem, b_shmem, acc, is_async=ASYNC)
 
     if ASYNC:
-        hopper.warpgroup_mma_wait(num_outstanding=1, deps=[acc])
+        acc = hopper.warpgroup_mma_wait(num_outstanding=1, deps=[acc])
```

Comment:
Do we have to actually use the result from the wait? I thought that the inline assembly would model the wait as a mutation, so llvm should already know not to reorder.

Reply:
If we do then I think updating the handle in-place might be a cleaner API.

Reply:
> Do we have to actually use the result from the wait? 

we do, we are only supposed to use the result from the wait

> I thought that the inline assembly would model the wait as a mutation, so llvm should already know not to reorder.

it models side-effects but here the wait needs ordering with pure operations so there is no way to model that in LLVM

Reply:
Yes, you actually have to use the wait. WGMMA is cursed and because the result is an SSA value, LLVM has no notion of side effects on what it thinks is an "immutable" SSA value. We have to thread the SSA value through the inline asm instruction to prevent LLVM from hoisting what it things are pure ops.